### DISCUSSION TITLE
Missing `.toHaveLabelText` and `.toHaveRole`?

### DISCUSSION THREAD

**Discussion by SevenOutman** [testing-library/jest-dom#424] [General]
Hi, I have been always wondering what is the correct way to test that an element has a specific ARIA role (explicitly or implicitly).

Say I'm implementing this `<Radio>` component which is expected to render an ARIA `radio` along with a `label` associated with it. Yes I can use `getByRole('radio')` to check that a `radio` is rendered, but I can't tell whether it's got the expected label text. And yes I can use `getByLabelText('text')` to check that something with expected label text is rendered, but I can't tell whether it's a `radio`.

Here's what I can do so far:

```js
expect(getByLabelText('text')).toEqual(getByRole('radio'));
```

Just as I was writing this comment I realized that I can do this:

```js
expect(getByRole('radio', { name: 'text' }));
```

But still posting this in case someone has same question with me.

**Comment by gnapse** [comment]
[`toHaveAccessibleName`](https://github.com/testing-library/jest-dom/#tohaveaccessiblename) can also be of help.

**Reply by SevenOutman** [reply]
For sure. I was just considering something pairing with `getByLabelText` and `getByRole` for consistency.
